src/SFA.DAS.AdminService.Web/Validators/Roatp/RoatpGatewayApplicationViewModelValidator.cs

```diff
@@ -0,0 +1,96 @@
+﻿using SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.AdminService.Web.Validators.Roatp
+{
+    public class RoatpGatewayApplicationViewModelValidator : IRoatpGatewayApplicationViewModelValidator
```

Comment:
There is a lot of repetition in this class and it is also very similar to RoatpGatewayPageViewModelValidator.  Could you spend half an hour and see if any reuse can be abstracted out?  I can't think how off the top of my head but definitely worth consideration.

Reply:
Yes. I've thought already about it. Although it follows similar logic there are too many differences.
I was tempted to make a custom text area validation attribute, but then in such case again the validation has to be mixed with radio buttons selection, which will require using validator.